### DISCUSSION TITLE
Not able to verify smart contract on zkSync Era Testnet using Remix and MetaMask

### DISCUSSION THREAD

**Discussion by vishalsingh2972** [Cyfrin/foundry-full-course-cu#2280] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/ce85ff4a-2abf-4631-828d-c2f0acd39cf8)
I am trying to deploy my smart contract on L2 but the "verify contract" button is disabled and even after deploying my smart contract is not visible on sepolia.explorer.zksync.io, any idea how to resolve this?

I even tried to verify my contract directly on sepolia.explorer.zksync.io, but again I am getting this error.
![image](https://github.com/user-attachments/assets/4ab672da-0c4c-4e13-8b0a-503b18cfcee0)

**Comment by EngrPips** [comment]
Hello @vishalsingh2972, Everything works fine for me. I was able to `deploy and verify` my contract and it also shows up in `sepolia.explorer.zksync.io` 

<img width="1440" alt="Screenshot 2024-08-20 at 20 57 53" src="https://github.com/user-attachments/assets/9204460c-5451-44be-b019-61231c77d1be">

<img width="1440" alt="Screenshot 2024-08-20 at 20 58 17" src="https://github.com/user-attachments/assets/569e973c-f1fb-4c60-99af-7c3a0812e916">

<img width="1440" alt="Screenshot 2024-08-20 at 21 02 02" src="https://github.com/user-attachments/assets/9e0878d2-b6f3-474a-aa27-c70fd9df0b31">

**Comment by EngrPips** [comment]
Did you actually check the `verify` box when you try to deploy? checking the box will give you access to use the `deploy and verify` function which will verify your contract on deployment and make it visible on `sepolia.explorer.zksync.io`

<img width="1440" alt="Screenshot 2024-08-20 at 21 04 06" src="https://github.com/user-attachments/assets/8f06c933-6076-4c7b-af80-2d625b3f1625">

**Reply by vishalsingh2972** [reply]
yes tried the exact same thing but the "verify" option is disabled and not able to select it, I'll retry once again after some time and update you.

**Reply by EngrPips** [reply]
Alright, Let's know how that plays out.

**Reply by vishalsingh2972** [reply, marked as answer]
![image](https://github.com/user-attachments/assets/1629588a-a1b5-455e-a6da-62504b704c32)
 update : there was some issue with the remix website, I was finally able to do it after 2 days. But thank you so much for replying so fast on discord as well as on github discussions @EngrPips. Cyfrin team is awesome.

**Reply by EngrPips** [reply]
You are welcome @vishalsingh2972, Keep crushing it.
